pkgs/checks/test/extensions/async_test.dart

```diff
@@ -15,10 +16,10 @@ void main() {
     test('completes', () async {
       (await checkThat(_futureSuccess()).completes()).equals(42);
 
-      await _rejectionWhichCheck(
+      await _rejectionWhichCheck<Future>(
         _futureFail(),
-        (f) async => await f.completes(),
-        (check) => check.single.equals('Threw UnimplementedError'),
+        it()..completes(),
```

Comment:
You can't easily `await` anymore, because of the cascade.

In many cases I don't think that should matter, but I'm not sure it will never matter.

Reply:
This seems to work OK using the async `.that` support. The replay will work correctly and the expectations are all checked.